internal/bminventory/inventory.go

```diff
@@ -4486,7 +4488,7 @@ func (b *bareMetalInventory) ListInfraEnvs(ctx context.Context, params installer
 }
 
 func (b *bareMetalInventory) RegisterInfraEnv(ctx context.Context, params installer.RegisterInfraEnvParams) middleware.Responder {
-	i, err := b.RegisterInfraEnvInternal(ctx, nil, params)
+	i, err := b.RegisterInfraEnvInternal(ctx, nil, nil, params)
```

Comment:
The ask came from Sylva so I think this is just his first pass: to get it working for kube api first. Will let Elior respond when he returns

Reply:
The current requirement is to enable this feature only for the Kube API. As @CrystalChun mentioned, this requirement originated from Sylva, and there is no need to implement it for the REST API. 
I discussed this with @avishayt and @gamli75 when I started working on the task, and since the only demand for this feature comes from Sylva, we decided not to implement it for the REST API.

Reply:
Okay no problem, was just curious.